enterprise/frontend/src/metabase-enterprise/remote_sync/components/SharedTenantCollectionsList/SharedTenantCollectionsList.tsx

```diff
@@ -0,0 +1,137 @@
+  const prevTypeRef = useRef(values[TYPE_KEY]);
+
+  // Reset collections to initial values when switching from read-write to read-only
+  useEffect(() => {
+    const prevType = prevTypeRef.current;
+    const currentType = values[TYPE_KEY];
+
+    if (prevType === "read-write" && currentType === "read-only") {
+      setFieldValue(COLLECTIONS_KEY, initialValues[COLLECTIONS_KEY] ?? {});
+    }
+
+    prevTypeRef.current = currentType;
+  }, [values, initialValues, setFieldValue]);
+
+  const handleToggle = useCallback(
+    (collection: CollectionItem, checked: boolean) => {
+      // Update form state directly via Formik context
+      setFieldValue(COLLECTIONS_KEY, {
+        ...syncMap,
+        [collection.id]: checked,
+      });
+    },
+    [syncMap, setFieldValue],
+  );
+
+  if (isLoading) {
+    return (
+      <Flex justify="center" py="lg">
+        <Loader data-testid="loading-indicator" />
+      </Flex>
+    );
+  }
+
+  if (error) {
+    return <Text c="error">{t`Failed to load shared tenant collections`}</Text>;
+  }
+
+  if (collections.length === 0) {
+    return <Text c="text-medium">{t`No shared tenant collections found`}</Text>;
+  }
+
+  return (
+    <Box
+      style={{
+        borderRadius: "var(--mantine-radius-md)",
+        border: "1px solid var(--mb-color-border)",
+        backgroundColor: "var(--mb-color-bg-white)",
+        overflow: "hidden",
+      }}
+    >
+      {collections.map((collection, index) => (
+        <CollectionSyncRow
+          key={collection.id}
+          collection={collection}
+          isChecked={syncMap[collection.id] ?? false}
+          onToggle={handleToggle}
+          isLast={index === collections.length - 1}
+          isReadOnly={isReadOnly}
+        />
+      ))}
+    </Box>
+  );
+};
+
+interface CollectionSyncRowProps {
+  collection: CollectionItem;
+  isChecked: boolean;
+  onToggle: (collection: CollectionItem, checked: boolean) => void;
+  isLast: boolean;
+  isReadOnly: boolean;
+}
+
+const CollectionSyncRow = ({
+  collection,
+  isChecked,
+  onToggle,
+  isLast,
+  isReadOnly,
+}: CollectionSyncRowProps) => {
+  const canWrite = collection.can_write ?? false;
+  const icon = PLUGIN_COLLECTIONS.getIcon({
+    model: "collection",
+    is_remote_synced: isChecked,
+  });
+
+  return (
+    <Box
+      p="md"
+      style={{
+        borderBottom: isLast ? undefined : "1px solid var(--mb-color-border)",
+      }}
+    >
+      <Flex justify="space-between" align="center">
+        <Flex align="center" gap="sm">
+          <Icon name={icon.name} c={icon.color ?? "text-medium"} />
+          <Text fw={500}>{collection.name}</Text>
+        </Flex>
+        <Flex align="center" gap="sm">
+          <Switch
+            size="sm"
+            checked={isChecked}
+            onChange={(e) => onToggle(collection, e.currentTarget.checked)}
+            disabled={!canWrite || isReadOnly}
+            aria-label={t`Sync ${collection.name}`}
```

Comment:
this string could use context, it's gonna be hard to translate `sync {0}` without context in a lot of language

```
c("{0} is the name of a metabase collection").t(`Sync ${collection.name}`)